### ISSUE TITLE
[BUG] Stuck at yarn build when using latest yarn pnp 

### ISSUE THREAD

**Issue by zigang93** [vercel/next.js#34794]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```
    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T8101
    Binaries:
      Node: 16.13.2
      npm: 8.1.2
      Yarn: 3.2.0
      pnpm: N/A
    Relevant packages:
      next: 12.1.0
      react: 17.0.2
      react-dom: 17.0.2
```

### What browser are you using? (if relevant)

Chrome 

### Describe the Bug

with the latest release of [yarn berry v3.2.0](https://github.com/yarnpkg/berry), yarn build seem like stuck forever at `info  - Checking validity of types`..

here is my reproduce [repo](https://github.com/zigang93/nextjs-pnp)

### Expected Behavior

Expect `yarn build`  to work well.

### To Reproduce

here is my reproduce [repo](https://github.com/zigang93/nextjs-pnp)
run `yarn install` first 
then `yarn build`

**Comment by ijjk** [comment]
Hi, updating to Node.js to >= `v16.14` should resolve this, related https://github.com/vercel/next.js/pull/34688#issuecomment-1047994505 we are investigating adding a proper error for this with the latest Yarn berry version.

**Comment by zigang93** [comment]
@ijjk solved with node 16.14 version

**Comment by pranayk2610** [comment]
Had the same problem in dockerfile using latest yarn version, thanks
